src/WebdamFolderPermissions.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace Drupal\media_webdam;
+
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\media_webdam\WebdamInterface;
+
+class WebdamFolderPermissions implements ContainerInjectionInterface {
+
+  use StringTranslationTrait;
+
+  protected $config_factory;
+  protected $webdam;
+
+  public function __construct(ConfigFactoryInterface $config_factory, WebdamInterface $webdam) {
+    $this->config = $config_factory->get('media_webdam.settings');
+    $this->webdam = $webdam;
+  }
+
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('config.factory'),
+      $container->get('media_webdam.webdam')
+    );
+  }
+
+  public function getPermissions() {
+    // Get list of folders on Webdam.
+    $webdamFolders = $this->webdam->getFlattenedFolderList();
```

Comment:
This should look at the folders that are enabled on the admin form. If there aren't any, then don't create any permissions at all.